internal/jobs/launch_instance_aws.go

```diff
@@ -15,38 +15,50 @@ import (
 )
 
 type LaunchInstanceAWSTaskArgs struct {
-	AccountID     int64 `json:"account_id"`
-	PubkeyID      int64 `json:"pubkey_id"`
+	// Associated reservation
 	ReservationID int64 `json:"reservation_id"`
-	// Optional name, can be blank
-	Name string `json:"name"`
-	// AWS AMI
+
+	// Associated account
+	AccountID int64 `json:"account_id"`
+
+	// Associated public key
+	PubkeyID int64 `json:"pubkey_id"`
+
+	// Detail information
+	Detail *models.AWSDetail `json:"detail"`
+
+	// AWS AMI as fetched from image builder
 	AMI string `json:"ami"`
-	// Amount of instances to launch
-	Amount int32 `json:"amount"`
-	// Immediately power off the system
-	PowerOff bool `json:"poweroff"`
-	// Amazon EC2 Instance Type
-	InstanceType string `json:"instance_type"`
+
 	// The ARN fetched from Sources which is linked to a specific source
 	ARN string `json:"arn"`
 }
 
+// Unmarshall arguments and handle error
 func HandleLaunchInstanceAWS(ctx context.Context, job dejq.Job) error {
+	args := LaunchInstanceAWSTaskArgs{}
+	decodeJob(ctx, job, args)
```

Comment:
this allows to continue the execution even if the decoding has failed, is that what we want?

Reply:
No, good point. Added a return statements at caller sites.